model-optimizer/mo/front/caffe/loader.py

```diff
@@ -280,6 +280,7 @@ def caffe_pb_to_nx(graph, proto, model):
                 input_dims.append(np.array(list(dims), dtype=np.int64))
                 input_names.append(layer.name)
 
+        layer_name = layer.name
         layer.name = graph.unique_id(layer.name)
```

Comment:
We should not modify the `layer` object. Instead I recommend to remove `layer_name` variable and use the following approach:

```
node_id = graph.unique_id(layer.name)
graph.add_node(node_id, pb=layer, model_pb=model_layer, kind='op', type='Parameter')
```

and update the places where `layer.name` is used in context of our graph node (line 289 and 296) to use `node_id` variable.

Reply:
Also in line 310 we use `input_names` list containing original `layer.name` values. But we probably should use the corresponding `node_id` there. Please, double check.

Reply:
Done.